examples/vision/siamese.py

```diff
@@ -0,0 +1,345 @@
+
+"""
+# Siamese network with a contrastive loss
+
+Author: Mehdi<br>
+Date created: 2021/05/06<br>
+Last modified: 2020/05/06<br>
+Description: Similarity learning using siamese network with contrastive loss
+"""
+
+"""
+## Introduction
+
+[Siamese Network](https://en.wikipedia.org/wiki/Siamese_neural_network)
+is any Neural Network which share weights between two or more sister networks,
+each producing embedding vector of its respective input and these embeddings
+are then passed through some 
+[distance heuristic](https://developers.google.com/machine-learning/clustering/similarity/measuring-similarity)
+to find the distance between them. This distance is later used to increase the
+contrast between embeddings of inputs of different classes and decrease it with
+that of similar class by employing some loss function, with the main objective
+of contrasting [vector spaces](https://en.wikipedia.org/wiki/Vector_space)
+from which these sample inputs were taken.
+
+"""
+
+"""
+## Setup
+"""
+
+import random
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras.models import Model
```

Comment:
Just `from tensorflow import keras` then access objects from the `keras.` namespace, e.g. `keras.Model`